os/arch/arm/src/s5j/sss/security_hal_wrapper.c

```diff
@@ -0,0 +1,1632 @@
+		break;
+	case HAL_KEY_DH_1024:
+	case HAL_KEY_DH_2048:
+	case HAL_KEY_DH_4096:
+		*key_type = SECURE_STORAGE_TYPE_KEY_DH;
+		break;
+	default:
+		return HAL_INVALID_ARGS;
+	}
+
+	return HAL_SUCCESS;
+}
+
+int hal_remove_key(hal_key_type mode, uint32_t key_idx)
+{
+	HWRAP_ENTER;
+	unsigned int key_type;
+	uint32_t ret;
+
+	ret = hal_get_key_type(mode, &key_type);
+	if (ret != HAL_SUCCESS) {
+		return ret;
+	}
+	ret = isp_remove_key(key_type, key_idx);
+	if (ret != 0) {
+		isp_clear(0);
+		printf("ISP failed (%zu)\n", ret);
+		return HAL_FAIL;
+	}
+	return HAL_SUCCESS;
+}
+
+int hal_generate_key(hal_key_type mode, uint32_t key_idx)
+{
+	HWRAP_ENTER;
+	uint32_t ret;
+	ISP_CHECKBUSY();
+	switch (mode) {
+	case HAL_KEY_AES_128:
+		ret = isp_aes_generate_key_securekey(16, key_idx);
+		break;
+	case HAL_KEY_AES_192:
+		ret = isp_aes_generate_key_securekey(24, key_idx);
+		break;
+	case HAL_KEY_AES_256:
+		ret = isp_aes_generate_key_securekey(32, key_idx);
+		break;
+	case HAL_KEY_RSA_1024:
+		ret = isp_rsa_generate_key_securekey(key_idx, 0xB1, 0);
+		break;
+	case HAL_KEY_RSA_2048:
+		ret = isp_rsa_generate_key_securekey(key_idx, 0xB2, 0);
+		break;
+	case HAL_KEY_RSA_3072:
+		ret = isp_rsa_generate_key_securekey(key_idx, 0xB3, 0);
+		break;
+	case HAL_KEY_RSA_4096:
+		return HAL_NOT_SUPPORTED;
+	case HAL_KEY_ECC_BRAINPOOL_P256R1:
+		ret = isp_ecdsa_generate_key_securekey(key_idx, 0x53);
+		break;
+	case HAL_KEY_ECC_BRAINPOOL_P384R1:
+		return HAL_NOT_SUPPORTED;
+		break;
+	case HAL_KEY_ECC_BRAINPOOL_P512R1:
+		return HAL_NOT_SUPPORTED;
+		break;
+	case HAL_KEY_ECC_SEC_P192R1:
+		ret = isp_ecdsa_generate_key_securekey(key_idx, 0x21);
+		break;
+	case HAL_KEY_ECC_SEC_P224R1:
+		ret = isp_ecdsa_generate_key_securekey(key_idx, 0x22);
+		break;
+	case HAL_KEY_ECC_SEC_P256R1:
+		ret = isp_ecdsa_generate_key_securekey(key_idx, 0x23);
+		break;
+	case HAL_KEY_ECC_SEC_P384R1:
+		ret = isp_ecdsa_generate_key_securekey(key_idx, 0x24);
+		break;
+	case HAL_KEY_ECC_SEC_P512R1:
+		ret = isp_ecdsa_generate_key_securekey(key_idx, 0x25);
+		break;
+	case HAL_KEY_HMAC_MD5:
+	case HAL_KEY_HMAC_SHA1:
+		ret = isp_hmac_generate_key_securekey(64, key_idx);
+		break;
+	case HAL_KEY_HMAC_SHA224:
+		ret = isp_hmac_generate_key_securekey(64, key_idx);
+		break;
+	case HAL_KEY_HMAC_SHA256:
+		ret = isp_hmac_generate_key_securekey(128, key_idx);
+		break;
+	case HAL_KEY_HMAC_SHA384:
+		ret = isp_hmac_generate_key_securekey(128, key_idx);
+		break;
+	case HAL_KEY_HMAC_SHA512:
+		ret = isp_hmac_generate_key_securekey(64, key_idx);
+		break;
+	default:
+		return HAL_NOT_SUPPORTED;
+	}
+	if (ret != 0) {
+		isp_clear(0);
+		printf("ISP failed (%zu)\n", ret);
+		return HAL_FAIL;
+	}
+
+	return HAL_SUCCESS;
+}
+
+/**
+ * Authenticate
+ */
+
+int hal_generate_random(uint32_t len, hal_data *random)
+{
+	HWRAP_ENTER;
+	uint32_t ret;
+	unsigned int inbuf[HAL_MAX_RANDOM_SIZE];
+
+	if (len > HAL_MAX_RANDOM_SIZE) {
+		return HAL_INVALID_ARGS;
+	}
+
+	if (len & 0x3) {
```

Comment:
I mean code like this.
Here it calculates with constant value, it is very hard to understand.

Reply:
This format is provided by DS isp code, which we do not need to take care. Thanks for the comment.